modules/RTC/RTCUtils.js

```diff
@@ -1041,12 +985,32 @@ class RTCUtils extends Listenable {
                     RTCEvents.DEVICE_LIST_AVAILABLE,
                     availableDevices);
 
-                if (isDeviceChangeEventSupported) {
+                if (browser.supportsDeviceChangeEvent()) {
                     navigator.mediaDevices.addEventListener(
                         'devicechange',
                         () => this.enumerateDevices(onMediaDevicesListChanged));
                 } else {
-                    pollForAvailableMediaDevices();
+                    // Periodically poll enumerateDevices() method to check if
+                    // list of media devices has changed.
+                    const pollMediaDevices = () => {
+                        this.enumerateDevices(pds => {
+                            // We don't fire RTCEvents.DEVICE_LIST_CHANGED for
+                            // the first time we call enumerateDevices().
+                            // This is the initial step.
+                            if (typeof availableDevices === 'undefined') {
+                                availableDevices = pds.slice(0);
+                            } else if (compareAvailableMediaDevices(pds)) {
+                                onMediaDevicesListChanged(pds);
+                            }
+
+                            availableDevicesPollTimer
+                                = window.setTimeout(
```

Comment:
Why prefer setTimeout over setInterval?

Reply:
Nice catch! No reason, I was moving the code around. Come to think of it, it makes more sense. I'll update it, please don't merge it, or feel free to amend my commit or add one of your own with the change. 👍

Reply:
Okay, up to you. I noticed soon after writing the question that the original was using a setTimeout; I'm not expecting you to change it. The reason why I haven't merged yet is because there's a (flaky) test failure.